### PR TITLE
HIP-584: Provide implementation for ExchangeRatePrecompiledContract

### LINKED ISSUE #6606: HIP-584: Provide implementation for ExchangeRatePrecompiledContract
- Copy ExchangeRatePrecompiledContract in com.hedera.services.
- Integrate ExchangeRatePrecompiledContract with the Store interface.
- Add Javadoc describing what was changed in the mirror-node code base
- Create integration tests in ContractCallServicePrecompileTest in hedera-mirror-web3 and validate the returned gas estimates are in the desired range between 5% and 20% more than the actual gas used

### PR DESCRIPTION
**Pull request by mustafauzunn**
**Description**:

Provide implementation for ExchangeRatePrecompiledContract

**Related issue(s)**:

Fixes #6606 

**Notes for reviewer**:

**Checklist**

- [ ] Documented (Code comments, README, etc.)
- [x] Tested (unit, integration, etc.)

### CODE REVIEW

File: hedera-mirror-web3/src/main/java/com/hedera/services/fees/HbarCentExchange.java [hunk 1]
```diff
@@ -18,11 +18,15 @@
 
 import com.hederahashgraph.api.proto.java.ExchangeRate;
 import com.hederahashgraph.api.proto.java.Timestamp;
+import java.time.Instant;
 
 /**
  *  Copied Logic type from hedera-services. Differences with the original:
  *  1. Remove unused methods: activeRate, activeRates, updateRates, fcActiveRates
```
Comment (steven-sheehy) [inline hedera-mirror-web3/src/main/java/com/hedera/services/fees/HbarCentExchange.java:25]: Update javadoc

File: hedera-mirror-web3/src/main/java/com/hedera/services/fees/BasicHbarCentExchange.java [hunk 2]
```diff
@@ -34,6 +35,14 @@
 public final class BasicHbarCentExchange implements HbarCentExchange {
     private final RatesAndFeesLoader ratesAndFeesLoader;
 
+    @Override
+    public ExchangeRate activeRate(final Instant now) {
```
Comment (steven-sheehy) [inline hedera-mirror-web3/src/main/java/com/hedera/services/fees/BasicHbarCentExchange.java:39]: Update class javadoc

File: hedera-mirror-web3/src/main/java/com/hedera/mirror/web3/evm/properties/MirrorNodeEvmProperties.java [hunk 3]
```diff
@@ -58,6 +58,8 @@ public class MirrorNodeEvmProperties implements EvmProperties {
 
     private boolean dynamicEvmVersion = true;
 
+    private long exchangeRateGasReq = 100;
```
Comment (steven-sheehy) [inline hedera-mirror-web3/src/main/java/com/hedera/mirror/web3/evm/properties/MirrorNodeEvmProperties.java:61]: Also add to docs/configuration.md.

```suggestion
    @Min(1)
    private long exchangeRateGasReq = 100;
```

**Review by steven-sheehy — CHANGES_REQUESTED** [review]

File: docs/configuration.md [hunk 4]
```diff
@@ -607,3 +607,4 @@ value, it is recommended to only populate overridden properties in the custom `a
  `hedera.mirror.web3.evm.trace.enabled`                 | false                                              | Flag enabling tracer                                                                                                                                                                          
  `hedera.mirror.web3.evm.trace.contract`                | []                                                 | A set with contract addresses to filter. By default it is empty to indicate it will trace all contract addresses.                                                                             
  `hedera.mirror.web3.evm.trace.status`                  | []                                                 | A set with frame statuses to filter. By default it is empty to indicate it will trace all frames regardless of status.                                                                        
+ `hedera.mirror.web3.evm.exchangeRateGasReq`            | 100                                                | Gas requirement for ExchangeRatePrecompile.
```
Comment (steven-sheehy) [inline docs/configuration.md:610]: nit: sort

**Review by steven-sheehy — APPROVED** [review]
LGTM. Can merge with another signoff

**Review by 0xivanov — APPROVED** [review]
LGTM

**Review by steven-sheehy — APPROVED** [review]

**Pull request merged**